src/app/components/subtopics-list/subtopics-list.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { SubtopicComponent } from './subtopic/subtopic.component';
+import { TopicService } from '../../services/topic.service';
+import { SubTopic } from '../../models/subtopic.model';
+import { SearchSectionComponent } from './search-section/search-section.component';
+
+@Component({
+  selector: 'app-subtopics-list',
+  standalone: true,
+  imports: [CommonModule, SubtopicComponent, SearchSectionComponent],
+  providers: [TopicService],
+  templateUrl: './subtopics-list.component.html',
+  styleUrl: './subtopics-list.component.scss',
+})
+export class SubtopicsListComponent implements OnChanges {
+  @Input() topicName: string = '';
+  protected subtopics: SubTopic[] | undefined;
```

Comment:
Dont forget to remove  | undefined in here after you refactor in topic.service.ts. Hint: If you ever wanna use undefined as a value in a class property, just do it like this: 
`protected subtopics?: SubTopic[]`